### PR TITLE
Spanner: Aborted batch dml should be retried

### PR DESCRIPTION
**Pull request by olavloite**
When executing batchDml statements, it is possible that the transaction is aborted by Spanner for one of the statements. This used to throw a SpannerBatchUpdateException, which would cause the transaction to fail, while the appropriate exception is AbortedException to trigger a retry of the transaction.

### CODE REVIEW

File: google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/TransactionRunnerImpl.java [hunk 1]
```diff
@@ -270,7 +271,12 @@ public long executeUpdate(Statement statement) {
         results[i] = response.getResultSets(i).getStats().getRowCountExact();
       }
 
-      if (response.getStatus().getCode() != 0) {
+      // If one of the DML statements was aborted, we should throw an aborted exception.
```
Comment (kolea2) [inline google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/TransactionRunnerImpl.java]: is it possible to also add a test to TransactionRunnerImplTest?

Comment (olavloite) [reply google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/TransactionRunnerImpl.java]: Added.

File: google-cloud-clients/google-cloud-spanner/src/test/java/com/google/cloud/spanner/TransactionRunnerImplTest.java [hunk 2]
```diff
@@ -141,6 +154,82 @@ public void runResourceExhaustedNoRetry() throws Exception {
...
+                public long[] run(TransactionContext transaction) throws Exception {
+                  numCalls.incrementAndGet();
+                  return transaction.batchUpdate(Arrays.asList(statement, statement));
+                }
+              });
+    } catch (SpannerBatchUpdateException e) {
+      exception = e;
+      updateCount = e.getUpdateCounts();
+    }
+    if (status == Code.ABORTED_VALUE) {
```
Comment (sduskis) [inline google-cloud-clients/google-cloud-spanner/src/test/java/com/google/cloud/spanner/TransactionRunnerImplTest.java]: Is there anyway to move the asserts into `batchDmlAborted` and `batchDmlFailedPrecondition`?

**Comment by kolea2** [comment]
@olavloite LGTM after running code formatting

**Pull request merged**